### PR TITLE
Add a view to allow configuring hostname.

### PR DESCRIPTION
**Pull request by mwhudson**
This seems like a lot of typing/copypaste for a very simple view... I didn't think very hard while doing this.

See bug https://bugs.launchpad.net/ubuntu/+source/subiquity/+bug/1621380

### CODE REVIEW

File: subiquitycore/ui/views/hostname.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    def cancel(self, button):
+        self.signal.prev_signal()
+
+    def done(self, button):
+        if len(self.hostname.value) < 1:
+            self.error.set_text("Please enter a non-empty name.")
+            return
+        if not self.opts.dry_run:
+            cmd = ["hostname", self.hostname.value]
```
Comment (mwhudson) [inline subiquitycore/ui/views/hostname.py]: Ah yeah, you're right. Is editing /etc/hosts really necessary though? After running hostname foo.local on in my KVM test system I can resolve foo.local. (Maybe this is some resolved magic?)

**Comment by mwhudson** [comment]
PTAL

**Comment by cyphermox** [comment]
I hadn't caught on to this, but applying the changes should go in the controller, not the view. I think we can also use hostnamectl to do the full change work, so maybe this is going to work better (I'll test).

**Comment by mwhudson** [comment]
Well after hacking around in the insides of subiquity a bit, here's another go. I like this one more :-)

I think /perhaps/ using hostnamectl would be better, but it's hard to be certain. Let me know what you think!

**Comment by mwhudson** [comment]
Ping?

**Comment by cyphermox** [comment]
There's a question as to whether this really does make sense to have for console-conf; maybe it shouldn't be a feature there. It still makes sense in the world of subiquity, but requiring people to configure the hostname of the device seems like one more step that people would have to do before ownership, whereas they could just as well do it later -- it's not required for the device to be reachable and usable.

**Comment by xnox** [comment]
From the bug report

> The DHCP server might have told you to use a hostname to use when you're assigned an address. If so, default to that in the prompt to the user.

I don't think we do this today, so I guess we should probe hostnamectl for the epehemral hostname and present that in subiquity UI too.

**Comment by mwhudson** [comment]
Closing ancient pull requests.

**Pull request closed without merging**